src/generator.cs

```diff
@@ -6241,19 +6272,35 @@ public void Generate (Type type)
 								library_name = CoreServicesMap;
 								break;
 							}
-						} 
+						} else {
+							// we get something in LibraryName from FieldAttribute so we asume
+							// it is a path to a library, so we save the path and change library name
+							// to a valid identifier if needed
+							library_path = library_name;
+							if (library_name.Contains ("/")) {
+								var lastSlashIdx = library_name.LastIndexOf ("/");
+								library_name = library_name.Substring (lastSlashIdx + 1);
```

Comment:
@dalexsoto any reason why `Path.GetFileName` would not work to replace those 3 lines ?